packages/@react-spectrum/numberfield/src/NumberField.tsx

```diff
@@ -10,85 +10,163 @@
  * governing permissions and limitations under the License.
  */
 
-import {classNames, useStyleProps} from '@react-spectrum/utils';
+import {AriaButtonProps} from '@react-types/button';
+import {classNames, useFocusableRef, useStyleProps} from '@react-spectrum/utils';
+import {Field} from '@react-spectrum/label';
+import {FocusableRef} from '@react-types/shared';
 import {FocusRing} from '@react-aria/focus';
 import inputgroupStyles from '@adobe/spectrum-css-temp/components/inputgroup/vars.css';
-import React, {RefObject, useRef} from 'react';
+import {mergeProps} from '@react-aria/utils';
+import React, {HTMLAttributes, InputHTMLAttributes, RefObject, useRef} from 'react';
 import {SpectrumNumberFieldProps} from '@react-types/numberfield';
-import StepButton from './StepButton';
+import {StepButton} from './StepButton';
 import stepperStyle from '@adobe/spectrum-css-temp/components/stepper/vars.css';
 import {TextFieldBase} from '@react-spectrum/textfield';
+import {useFormProps} from '@react-spectrum/form';
+import {useHover} from '@react-aria/interactions';
 import {useNumberField} from '@react-aria/numberfield';
 import {useNumberFieldState} from '@react-stately/numberfield';
-import {useProviderProps} from '@react-spectrum/provider';
+import {useProvider, useProviderProps} from '@react-spectrum/provider';
 
-
-export const NumberField = React.forwardRef((props: SpectrumNumberFieldProps, ref: RefObject<HTMLDivElement>) => {
+function NumberField(props: SpectrumNumberFieldProps, ref: FocusableRef<HTMLElement>) {
   props = useProviderProps(props);
+  props = useFormProps(props);
+  let provider = useProvider();
   let {
-    // formatOptions,
     isQuiet,
     isDisabled,
-    showStepper = true,
-    autoFocus,
-    ...otherProps
+    hideStepper
   } = props;
-  let {styleProps} = useStyleProps(props);
-  let state = useNumberFieldState(otherProps);
+  let fieldProps = Object.assign({}, props);
+  // TS won't remove onChange from the type, so when using fieldProps below, cast it as omitted
+  delete fieldProps['onChange'];
```

Comment:
i mean, it doesn't have to be, but i do still need to Omit it because 
```
 Types of property 'onChange' are incompatible. 
 Type '(value: number) => void' is not assignable to type '(event: FormEvent<HTMLElement>) => void'.
 Types of parameters 'value' and 'event' are incompatible.
 Type 'FormEvent<HTMLElement>' is not assignable to type 'number'.
```